apps/brandfolder/src/index.js

```diff
@@ -0,0 +1,93 @@
+import { setup } from 'shared-dam-app';
+
+const CTF_APP_URL = 'https://brandfolder.ctfapps.net'
+const BF_EMBED_URL = `https://integration-panel-ui.brandfolder-svc.com?channel=message&appName=Contentful&origin=${CTF_APP_URL}&initMsg=hi`
+
+
+const CTA = 'Select an asset on Brandfolder';
+
+function makeThumbnail(attachment) {
+  const thumbnail = attachment.thumbnail_url || attachment.url
+  const url = typeof thumbnail === 'string' ? thumbnail : undefined
+  const alt = attachment.filename
+
+  return [url, alt];
+}
+
+
+function renderDialog(sdk) {
+  const config = sdk.parameters.invocation;
+
+  const container = document.createElement('div');
+  container.innerHTML = `<iframe id='brandfolder-embed' class='iframe-container' src='${BF_EMBED_URL}' width=400 height=650 style='border:none;'/>`;
+  document.body.appendChild(container);
+
+  sdk.window.startAutoResizer();
+
+  window.addEventListener('message', e => {
+    const { data, origin } = e
+    if (origin === 'https://integration-panel-ui.brandfolder-svc.com'){
+      const { event, payload } = data;
+      if (event === 'selectedAttachment') {
+        sdk.close([payload]);
+      }
+      else if (data.event === 'selectedAsset' && payload.attachments.length !== 0) {
+        const att_id = payload.attachments[0].id
+        const attachment = payload.included.find(att => att.id === att_id)
+        if (attachment){
+          sdk.close([attachment])
+        }
+      }
+    }
+  });
+}
+
+async function openDialog(sdk, _currentValue, config) {
+  const result = await sdk.dialogs.openCurrentApp({
+    position: 'center',
+    title: CTA,
+    shouldCloseOnOverlayClick: true,
+    shouldCloseOnEscapePress: true,
+    parameters: { ...config },
+    width: 400,
+    allowHeightOverflow: true,
+  });
+
+  if (!Array.isArray(result)) {
+    return [];
+  }
+
+  // Example result:
+  // [{
+  //     "id": "pfbfh7-f4zem0-dpcdo2",
+  //     "type": "attachments",
+  //     "mimetype": "image/png",
+  //     "filename": "brandfolder-icon.png",
+  //     "size": 9252,
+  //     "width": 312,
+  //     "height": 294,
+  //     "url": "https://s3.amazonaws.com/bf.boulder.prod/pfbfh7-f4zem0-dpcdo2/original/brandfolder-icon.png",
+  //     "thumbnail_url": "https://assets.brandfolder.com/pfbfh7-f4zem0-dpcdo2/element.png?v=1555005228",
+  //     "position": 0,
+  //     "included": null,
+  //     "asset": {
+  //         "id": "6skcfbkq"
+  //     }
+  // }]
+  return result
```

Comment:
For safety purposes (not to accidentally leak PII in the future) in all the other we are whitelisting the fields of the response that we want persisted on the Contentful side (e.g. https://github.com/contentful/apps/blob/master/apps/cloudinary/src/index.js#L12 if you check how the `FIELDS_TO_PERSIST` constant is used).

Could we implement the same check here?